### PR TITLE
feat: indent/indentScript option

### LINKED ISSUE #155: indent rule requires that <script> elements be indented
This is how I want to write our svelte components:

```svelte
<script>
const foo = "bar";
</script>
```

But if I enable the indent rule, it really wants me to do this:

```svelte
<script>
    const foo = "bar";
</script>
```

I can disable it using `{ ignoredNodes : [ "SvelteScriptElement" ] }` but it means that none of the indentation for any direct children will be checked, which is kind of a bummer. I'm sure this is a bit annoying of an ask, but would you be open to adding a new option where a user can define the level of indentation they want specifically for `<script>`s? It could default to `1`, so it wouldn't be a breaking change.

If you're open to this proposal I'd be thrilled to figure out how to send a PR for it.

### PR DESCRIPTION
**Pull request by tivac**
Allows for configuring whether the contents of `<script>` blocks should be indented a single tabstop or not. Defaults to `true` for back-compat.

Fixes #155

### CODE REVIEW

File: src/rules/indent-helpers/svelte.ts [hunk 1]
```diff
@@ -38,7 +43,7 @@ export function defineVisitor(context: IndentContext): NodeListener {
             node.children.filter(isNotEmptyTextNode),
             node.startTag,
             node.endTag,
-            1,
+            options.indentScript ? 1 : 0,
```
Comment (ota-meshi) [inline src/rules/indent-helpers/svelte.ts:46]: I don't think this change is necessary. Did we need this change?

**Review by ota-meshi — CHANGES_REQUESTED** [review]

**Comment by ota-meshi** [comment]
Thank you for this PR! I made one comment.

File: src/rules/indent-helpers/svelte.ts (hunk 1 above)
Comment (tivac) [reply src/rules/indent-helpers/svelte.ts:46]: Unclear, I'll test without tomorrow and see what happens.

**Review by JounQin — CHANGES_REQUESTED** [review]
A new config for `prettier` with this rule disabledd should be added IMO. @ota-meshi

**Comment by ota-meshi** [comment]
Normally, the fixture should be formatted. However, stylistic rule invalid tests and option tests may ignore the format.
If we want to ignore settings completely, we need to set it for each rule, `valid` and` invalid` directory. Also, we may want to ignore the format only partly.
I think it easier to use the ignore comment than use ignore settings.

**Comment by ota-meshi** [comment]
Did I misunderstand what you were saying? Are you talking about the shareable configuration provided by this plugin?

**Comment by JounQin** [comment]
@ota-meshi I mean add a new config in this plugin: `extends: ['plugin:svlete/prettier']` like `jsonc` plugin?

**Comment by JounQin** [comment]
> Did I misunderstand what you were saying? Are you talking about the shareable configuration provided by this plugin?

Exactly.

**Comment by ota-meshi** [comment]
Yeah. I think we need a new shareable configuration, as you say. We can work on it with another PR.

**Review by ota-meshi — APPROVED** [review]
LGTM! Thank you!

**Pull request merged**